### ISSUE TITLE
Got low AUC on cifar-10

### ISSUE THREAD

**Issue by JPerAsperaadAstra** [LucStrater/GeneralAD#2]
Hi dear authors, thanks for your great work and open-source codes. When I conduct the experiment on cifar-10 with released code, I got  a rather low AUC on test set after 20 epochs training using vit base model as the picture shows:

![image](https://github.com/user-attachments/assets/20c739f2-b8a2-4fa7-8fde-6d45d4c0080e)
My running command is: python main.py --wandb_entity xxx --wandb_api_key xxxx --wandb_name generalAD --run_type general_ad --epochs 20 --image_size 518 --batch_size 16
I noticed that in the paper the result on cifar-10 after 20 epochs training is 99.1, I wonder why is this happening and whether you could give me some suggestions on reproducing the results. Thanks

**Comment by LucStrater** [comment]
Thank you for your question and for your interest in our work!

To reproduce the results from the paper, you have two options:

1. Run the job file `GeneralAD/jobs/run_general_ad.job` and comment out all datasets except for CIFAR-10.

2. Use the parameters from the command line in the job file and run it separately for every class.

Given that you're aiming for the 99.1 AUC result, I assume you want to use the base variant rather than the large variant. To do this, you should change the `--hf_path` parameter to 'vit_base_patch14_dinov2.lvd142m'.

Here's an example command line for running the experiment on CIFAR-10 with the base model:

```bash
python main.py \
    --normal_class 0 \
    --epochs 20 \
    --batch_size 16 \
    --test_batch_size 16 \
    --lr 0.0005 \
    --lr_decay_factor 0.2 \
    --seed 0 \
    --hf_path 'vit_base_patch14_dinov2.lvd142m' \
    --image_size 518 \
    --layers_to_extract_from '12' \
    --hidden_dim 2048 \
    --noise_std 0.25 \
    --log_every_n_steps 4 \
    --run_type 'general_ad' \
    --dataset 'cifar10' \
    --wandb_entity your_wandb_entity \
    --wandb_api_key your_wandb_api_key \
    --wandb_name 'GeneralAD' \
    --data_dir 'data/' \
    --num_fake_patches -1 \
    --dsc_layers 1 \
    --dsc_heads 4 \
    --dsc_dropout 0.1 \
    --fake_feature_type 'base' \
    --top_k -1 \
    --smoothing_sigma 16 \
    --smoothing_radius 18 \
    --shots -1 \
    --val_monitor 'image_auroc' \
    --log_pixel_metrics 0
```

Make sure to replace `your_wandb_entity` and `your_wandb_api_key` with your actual Weights & Biases credentials.

You'll need to run this command for each class (0 through 9) by changing the `--normal_class` parameter.

If you're still experiencing issues after trying these suggestions, please let us know, and we'll be happy to help further.

**Comment by JPerAsperaadAstra** [comment]
Thx! Using the same parameter settings as above could produce the same auc result as that in the paper! Could you please give some suggestions about choosing the parameters? such as dsc_heads noise_std, etc. Since these parameters seem to be empirical.
![image](https://github.com/user-attachments/assets/f5380f62-ae3b-4dca-882a-be7c19a998ea)

**Comment by LucStrater** [comment]
I'm glad to hear that you were able to reproduce the results! Regarding parameter selection, I can provide some insights based on our experiments and ablation studies:

1. noise_std and top_k: These were the most critical parameters we tuned. For CIFAR-10, you can refer to Figure 4 in our paper. The performance is relatively stable for noise_std between 0.15 and 0.35.

2. dsc_heads: This parameter (number of heads in the discriminator) was primarily tuned on MVTec-LOCO. It's more important for datasets where we perform shuffling of features, as each head focuses on different image elements. For semantic datasets like CIFAR-10, where we only noise all patches without shuffling, this parameter doesn't significantly affect the image AUROC score (we tested between 1-16 heads).

3. dsc_layers: We found that multiple discriminator layers performed significantly worse, as the model started overfitting quite soon. That's why we settled on a single layer.

4. Learning rate and batch size: These were tuned together. For CIFAR-10, the model was very stable with learning rates between 0.0001 and 0.0007.

5. hidden_dim: This was primarily tuned for industrial datasets. For semantic datasets like CIFAR-10, higher values might be marginally better (we tested the range 1024-6144).

**Comment by JPerAsperaadAstra** [comment]
Thank you again for your patient in answering! No doubts left :)
